### PR TITLE
CIS Azure rules 9.1, 9.3, and 9.10

### PR DESCRIPTION
**Pull request by kubasobon**
### Summary of your changes

Adds 3 CIS Azure rules:
- **9.1** Ensure App Service Authentication is set up for apps in Azure App Service
- **9.3** Ensure Web App is using the latest version of TLS encryption
- **9.10** Ensure FTP deployments are Disabled

#### 9.1 - `.extension.authSettings.properties.Enabled == true`

image

#### 9.3 - `.extension.siteConfig.properties.MinTLSVersion == "1.2"`

image

#### 9.10 - `.extension.siteConfig.properties.FtpsState is "FtpsOnly" or "Disabled"`

image

### Related Issues

Fixes https://github.com/elastic/cloudbeat/issues/1758

### Checklist
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary README/documentation (if appropriate)

#### Introducing a new rule?

- [x] Generate rule metadata using [this script](https://github.com/elastic/cloudbeat/tree/main/security-policies/dev#generate-rules-metadata)
- [x] Add relevant unit tests
- [x] Generate relevant rule templates using [this script](https://github.com/elastic/cloudbeat/tree/main/security-policies/dev#generate-rule-templates), and open a PR in [elastic/packages/cloud_security_posture](https://github.com/elastic/integrations/tree/main/packages/cloud_security_posture)

### CODE REVIEW

File: security-policies/bundle/compliance/cis_azure/rules/cis_9_3/rule.rego [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	result := common.generate_result_without_expected(
+		common.calculate_result(is_min_tls_version_1_2),
+		data_adapter.resource,
+	)
+}
+
+default is_min_tls_version_1_2 := false
+
+is_min_tls_version_1_2 if {
+	data_adapter.resource.extension.siteConfig.properties.MinTLSVersion == "1.2"
```
Comment (oren-zohar) [inline security-policies/bundle/compliance/cis_azure/rules/cis_9_3/rule.rego:21]: Should be gte for forward compatibility IMO

File: security-policies/bundle/compliance/cis_azure/rules/cis_9_3/test.rego [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	eval_fail with input as test_data.generate_azure_asset_with_ext(kind, {}, {"siteConfig": {}})
+	eval_fail with input as test_data.generate_azure_asset_with_ext(kind, {}, {"siteConfig": {"properties": {}}})
+	eval_fail with input as test_data.generate_azure_asset_with_ext(kind, {}, {"siteConfig": {"properties": {"MinTLSVersion": null}}})
+	eval_fail with input as test_data.generate_azure_asset_with_ext(kind, {}, {"siteConfig": {"properties": {"MinTLSVersion": "1.0"}}})
+	eval_fail with input as test_data.generate_azure_asset_with_ext(kind, {}, {"siteConfig": {"properties": {"MinTLSVersion": "1.1"}}})
+}
+
+test_pass if {
+	eval_pass with input as test_data.generate_azure_asset_with_ext(kind, {}, {"siteConfig": {"properties": {"MinTLSVersion": "1.2"}}})
+}
```
Comment (oren-zohar) [inline security-policies/bundle/compliance/cis_azure/rules/cis_9_3/test.rego:23]: TLS 1.3 should pass

**Review by oren-zohar — CHANGES_REQUESTED** [review]

File: resources/providers/azurelib/inventory/app_service_provider.go [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+
+func defaultAzureAppServiceWrapper(credentials azcore.TokenCredential) *azureAppServiceWrapper {
+	return &azureAppServiceWrapper{
+		AssetWebAppsAuthSettings: func(ctx context.Context, subscriptionID, resourceGroupName, appName string) (armappservice.WebAppsClientGetAuthSettingsResponse, error) {
+			client, err := armappservice.NewWebAppsClient(subscriptionID, credentials, nil)
+			if err != nil {
+				return armappservice.WebAppsClientGetAuthSettingsResponse{}, err
+			}
+			response, err := client.GetAuthSettings(ctx, resourceGroupName, appName, nil)
+			if err != nil {
```
Comment (romulets) [inline resources/providers/azurelib/inventory/app_service_provider.go:44]: why handle the error and not just return `response, err`?

File: resources/providers/azurelib/inventory/app_service_provider.go [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+			}
+			return response, nil
+		},
+		AssetWebAppsSiteConfig: func(ctx context.Context, subscriptionID, resourceGroupName, appName string) (armappservice.WebAppsClientGetConfigurationResponse, error) {
+			client, err := armappservice.NewWebAppsClient(subscriptionID, credentials, nil)
+			if err != nil {
+				return armappservice.WebAppsClientGetConfigurationResponse{}, err
+			}
+			response, err := client.GetConfiguration(ctx, resourceGroupName, appName, nil)
+			if err != nil {
```
Comment (romulets) [inline resources/providers/azurelib/inventory/app_service_provider.go:55]: same question about error handling

**Review by romulets — APPROVED** [review]

File: resources/providers/azurelib/inventory/app_service_provider.go (hunk 3 above)
Comment (kubasobon) [reply resources/providers/azurelib/inventory/app_service_provider.go:44]: I have no strong preference one way or the other. If you tend to just return if types match, I can update the PR :)

**Comment by kubasobon** [comment]
@oren-zohar I'd appreciate your re-review :)

**Review by oren-zohar — APPROVED** [review]
LGTM 🚀

**Comment by kubasobon** [comment]
FYI: main saw a refactor of file structure, I rebased on top of it and force-pushed.

**Pull request merged**